Decred Wallet/Features/Navigation Menu/NavigationMenuBaseController.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  NavigationMenuBaseController.swift
+//  Decred Wallet
+//
+// Copyright (c) 2018-2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+import UIKit
+
+class NavigationMenuBaseController: UITabBarController {
+    
+    var isNewWallet: Bool = false
+    var floatingButtons: NavMenuFloatingButtons!
+    var customTabBar: TabMenu!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.loadTabBar()
+        self.loadFloatingButtons() // TODO: Organize this to only show on the two pages where it is needed
+        if self.isNewWallet {
+            self.showOkAlert(message: LocalizedStrings.newWalletMsg)
+        }
+    }
+    
+    func loadTabBar() {
+        let tabItems: [MenuItem] = [.overview, .transactions, .accounts, .more]
+        
+        self.setupCustomTabMenu(tabItems){ (controllers) in
+            self.viewControllers = controllers
+        }
+        self.selectedIndex = 0
+    }
+    
+    public func loadFloatingButtons() {
+        self.floatingButtons = NavMenuFloatingButtons()
+        self.floatingButtons.translatesAutoresizingMaskIntoConstraints = false
+        self.floatingButtons.clipsToBounds = true
+        
+        self.view.addSubview(self.floatingButtons)
+        let constraints = [
+            self.floatingButtons.heightAnchor.constraint(equalToConstant: 48),
+            self.floatingButtons.widthAnchor.constraint(equalToConstant: 240),
+            self.floatingButtons.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor),
+            self.floatingButtons.bottomAnchor.constraint(equalTo: customTabBar.topAnchor, constant: -12),
```

Comment:
Magic numbers. Use declared constants or add comments or do both.

Reply:
resolved in 37e2453